xarray/core/dataset.py

```diff
@@ -253,6 +253,11 @@ def _set_init_vars_and_dims(self, vars, coords, compat):
         """
         _assert_empty([k for k in vars if k in coords],
                       'redundant variables and coordinates: %s')
+
+        # Need to coerce Series to a dict as it doesn't implement __iter__ like a mapping
+        # GH470
+        vars = dict(vars) if isinstance(vars, pd.Series) else vars
```

Comment:
Can we do the two-liner instead? It's an extra line but less text -- and is clearer IMO